presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java

```diff
@@ -186,43 +192,13 @@ private String formatPlan(PlanNode plan, Map<Symbol, Type> types)
                 });
     }
 
-    private Rule.Context ruleContext(SymbolAllocator symbolAllocator, Lookup lookup, Session session)
-    {
-        return new Rule.Context()
-        {
-            @Override
-            public Lookup getLookup()
-            {
-                return lookup;
-            }
-
-            @Override
-            public PlanNodeIdAllocator getIdAllocator()
-            {
-                return idAllocator;
-            }
-
-            @Override
-            public SymbolAllocator getSymbolAllocator()
-            {
-                return symbolAllocator;
-            }
-
-            @Override
-            public Session getSession()
-            {
-                return session;
-            }
-        };
-    }
-
-    private static class RuleApplication
+    private static class OptimizationResult
```

Comment:
calling it "optimization" is not optimal, since rule are not, by definition, optimizations. Some are, but some aren't.
Previously  this class used to be called "RuleApplication" and I think this is much better name.

Also, considering class rename, please take into account changes that #8852 will make here.